src/CdrWriter.ts

```diff
@@ -166,6 +163,83 @@ export class CdrWriter {
     return this;
   }
 
+  /** Writes the delimiter header returning the endianness flag and object size
+   * NOTE: changing endian-ness with a single CDR message is not supported
```

Comment:
If this is not supported should `dHeader()` why even allow passing this option? The logic checks it against the value computed in the constructor and it seems weird to require the caller to pass in the matching value. I would remove this from the API surface area.